build-plugins/build-support/src/main/kotlin/aws/sdk/kotlin/gradle/plugins/artifactsizemetrics/AnalyzeArtifactSizeMetrics.kt

```diff
@@ -109,34 +110,73 @@ internal abstract class AnalyzeArtifactSizeMetrics : DefaultTask() {
         return ArtifactSizeMetricsAnalysis(artifactSizeMetrics, significantChange, changeHappened)
     }
 
+    // There are small fluctuations in artifact size that are not real delta
+    private fun Long.isNotaFluctuation() = abs(this) > 5L
+
     private data class ArtifactSizeMetricsAnalysis(
         val metrics: Map<String, ArtifactSizeMetric>,
         val significantChange: Boolean,
         val hasDelta: Boolean,
     )
 
-    private fun createDiffTable(analysis: ArtifactSizeMetricsAnalysis): String = buildString {
+    private fun createDiffTables(analysis: ArtifactSizeMetricsAnalysis): String = buildString {
         appendLine("Affected Artifacts\n=")
-        appendLine("| Artifact |Pull Request (bytes) | Latest Release (bytes) | Delta (bytes) | Delta (percentage) |")
-        appendLine("| -------- | ------------------: | ---------------------: | ------------: | -----------------: |")
-        analysis.metrics.forEach { metric ->
-            if (metric.value.delta != 0L) {
-                append("|")
-                append(metric.key)
-                append("|")
-                if (metric.value.currentSize == 0L) append("(does not exist)") else append("%,d".format(metric.value.currentSize))
-                append("|")
-                if (metric.value.latestReleaseSize == 0L) append("(does not exist)") else append("%,d".format(metric.value.latestReleaseSize))
-                append("|")
-                append("%,d".format(metric.value.delta))
-                append("|")
-                append("%.2f".format(metric.value.percentage))
-                append("%")
-                appendLine("|")
+
+        val requiresAttention = StringBuilder()
+            .appendLine("Significantly increased in size")
+            .appendLine()
+            .append(tableHeader)
+        var artifactRequiresAttention = false
+
+        val ordinary = StringBuilder()
+            .appendLine("<details>") // See: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections
+            .appendLine("<summary>Changed in size</summary>")
+            .appendLine()
+            .append(tableHeader)
+        var artifactOrdinaryChange = false
+
+        analysis.metrics
+            .toList()
+            .sortedBy { it.second.percentage }.toMap()
```

Comment:
**Nit**: This sorts in ascending order but I think we probably want _descending_ order.

**Style**: When using fluent call patterns, put each chained property access and method call on its own line.

```kotlin
analysis
    .metrics
    .toList()
    .sortedByDescending { it.second.percentage }
    .toMap()
    .forEach { metric ->
        …
```